src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs

```diff
@@ -115,6 +115,19 @@ private PackArgs GetPackArgs()
             return packArgs;
         }
 
+        private string[] GetTargetPathsToAssemblies()
+        {
+            if (TargetPathsToAssemblies == null)
+            {
+                return new string[0];
+            }
+
+            return TargetPathsToAssemblies
+                .Where(path => !string.IsNullOrEmpty(path))
```

Comment:
string.IsNullOrWhitespace might be safer since msbuild doesn't trim extra new lines characters